Add documentation for setting up a Self-Hosted Endpoint

greenberga opened the pull request.

jamielinux commented on docs/guides/self-hosted-endpoint.md:
```diff
@@ -0,0 +1,143 @@
+1. `X-Frc-Proxy-Key`: The proxy key you generated in the Friendly Captcha dashboard.
+2. `X-Frc-Proxy-Client-IP`: The original (source) IP address of the end user.
+
+As the upstream server for the proxied requests, you should specify
```
```suggestion
Forward proxied requests to this endpoint (i.e., the upstream server):
```

jamielinux commented on docs/guides/self-hosted-endpoint.md:
```diff
@@ -0,0 +1,143 @@
+
+### 3. Configure your site or app's widget
+
+The final step is to configure your server's URL as the API endpoint for your widget. This will ensure that the widget sends it request to your server (which will then forward them to the upstream Friendly Captcha API). If your server's URL is `https://organization.org` and you configure the widget using HTML `data-` attributes, your markup will look something like this:
```
```suggestion
The final step is to configure your server's URL as the API endpoint for your widget. This will ensure that the widget sends its requests to your server (which will then forward them to the upstream Friendly Captcha API). If your server's URL is `https://organization.org` and you configure the widget using HTML `data-` attributes, your markup will look something like this:
```

greenberga commented:
Glad I tested—I ended up making changes to the Apache, Caddy, _and_ HAProxy examples! But I've gotten them to all work smoothly. And, I think I've addressed all feedback thus far.

jamielinux commented on docs/guides/self-hosted-endpoint.md:
```diff
@@ -0,0 +1,146 @@
+
+1. Generate a proxy key in the Friendly Captcha dashboard.
+2. Configure your web server to proxy widget requests to the Friendly Captcha API.
+3. Configure your widget to use your Self-Hosted Endpoint.
```
I would maybe remove this chunk, or replace it with just `To set up a Self-Hosted Endpoint, follow the 3 steps below:` or similar.

Alternatively, it could be like a mini table of contents for the steps (where each bullet matches the wording of each heading and links to it).

In its current form, it's easy to miss that the bullet points match with the three steps below it.
